src-tauri/src/lib/settings.rs

```diff
@@ -515,4 +516,40 @@ impl Settings {
         using_existing_idf,
       })
     }
+
+    /// Compute the latency map for the tools mirror
+    pub async fn get_tools_mirror_latency_map(&self) -> Result<HashMap<String, u32>> {
+        let available_mirrors = crate::get_idf_tools_mirrors_list()
+            .to_vec()
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>();
+        let mirror_latency_map =
+            crate::utils::calculate_mirror_latency_map(&available_mirrors).await;
+        Ok(mirror_latency_map)
+    }
+
+    /// Compute the latency map for the IDF mirror
+    pub async fn get_idf_mirror_latency_map(&self) -> Result<HashMap<String, u32>> {
+        let available_mirrors = crate::get_idf_mirrors_list()
+            .to_vec()
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>();
+        let mirror_latency_map =
+            crate::utils::calculate_mirror_latency_map(&available_mirrors).await;
+        Ok(mirror_latency_map)
+    }
+
+    /// Compute the latency map for the PyPI mirror
+    pub async fn get_pypi_mirror_latency_map(&self) -> Result<HashMap<String, u32>> {
+        let available_mirrors = crate::get_pypi_mirrors_list()
+            .to_vec()
+            .iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<String>>();
+        let mirror_latency_map =
+            crate::utils::calculate_mirror_latency_map(&available_mirrors).await;
+        Ok(mirror_latency_map)
+    }
```

Comment:
@Hahihula I think i made it redundant by adding these functions here or I may have miss understood something